src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -132,11 +132,11 @@ def remove_id_events(dd):
                     out = [dict(o.to_json()) for o in wait(cephadm_module, c)]
                     expected = [
                         {
-                            'placement': {'count': 2},
+                            'placement': {'count': 1},
```

Comment:
I do not know why .. but the "with_daemon .... add_mds"  is not adding nothing to the describe_service output. Taking into account the change introduced "improve an output message" this is completely weird.

Reply:
found it: Your change:

![image](https://user-images.githubusercontent.com/2574405/109664684-ae3fca80-7b6d-11eb-8238-7bc817d5c29a.png)

affects this. previously the `if` did not match. now it matches.

Reply:
I think that the clause "and host"  is wrong because it causes not adding a placement spec when the service already does not have it and you does not provide a host.
In lot of tests that is exactly the situation we have. If we leave the "and host" then we have as result services deployed without placement. Now this problem is clearly visible because in this PR  we are "exposing" the placement as part of the output of the command.
With the modification proposed, if there is no placement spec defined in the service , a default new one is created, and if it is provided a host, the host is included in the new placement spec.

Another possibility... is to modify the "expected_result" and deem as valid services without placement spec.

But I continue without understand why the "daemon" added is not part of the mds service...